realm/realm-library/src/main/java/io/realm/internal/Table.java

```diff
@@ -904,8 +904,15 @@ private void invalidateCachedPrimaryKeyIndex() {
      * This will remove the prefix "class_" from all table names in the pk_column
      * Any database created on Realm-Java 0.84.1 and below will have this error.
      */
-    private void migratePrimaryKeyTableIfNeeded(long groupNativePtr, Table pkTable) {
-        nativeMigratePrimaryKeyTableIfNeeded(groupNativePtr, pkTable.nativePtr);
+    public static boolean migratePrimaryKeyTableIfNeeded(SharedRealm sharedRealm) {
+        if (sharedRealm == null || !sharedRealm.isInTransaction()) {
+            throwImmutable();
+        }
```

Comment:
`isImmutable` is an member function and used by lots of other classes... I will leave this is future's clean up PR.